### PR TITLE
Fix PRs running from forks

### PR DESCRIPTION
**Pull request by arturo-seijas**
Fixes the following actions when running from forks:
* Trivy scan
* Docs publishing in draft mode
* Lib checks

Runs:
* https://github.com/canonical/indico-operator/actions/runs/4143038228
* https://github.com/canonical/indico-operator/actions/runs/4143038232

### CODE REVIEW

**Review by nrobinaubertin — DISMISSED** [review]

File: .github/workflows/build_images.yaml [hunk 1]
```diff
@@ -78,25 +79,26 @@ jobs:
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
         if: ${{ github.event.pull_request.head.repo.fork }}
+      # Trivy requires and OCI exporter type to run the scan
```
Comment (nrobinaubertin) [inline .github/workflows/build_images.yaml:82]: s/and/an

**Review by amandahla — DISMISSED** [review]
Nice!

File: .github/workflows/build_images.yaml (hunk 1 above)
Comment (arturo-seijas) [reply .github/workflows/build_images.yaml]: Fixed, thx

**Review by nrobinaubertin — DISMISSED** [review]

**Review by nrobinaubertin — DISMISSED** [review]

**Review by weiiwang01 — DISMISSED** [review]

**Review by amandahla — APPROVED** [review]

**Review by mthaddon — APPROVED** [review]

**Comment by jdkandersson** [comment]
We should consider this carefully. It is beneficial for the CI-CD to be able to run for forks, although it isn't clear whether that should result in being able to merge into main. If we merge into main from a fork, anything that is disabled may have issues with it not uncovered by the disabled checks. An alternative is to require any PR forks to be merged into another branch on the repo, get it to run all the checks and then merge into main I'll write a note in the contributing guide where we can debate and link it here.

**Comment by jdkandersson** [comment]
It would also have been good to write a test for the operator workflows running in a fork. I'm not sure whether this would work, an initial idea is to use the GitHub cli to create a fork, trigger a PR and then query the status of that PR: https://cli.github.com/manual/gh_repo_fork

Without this test, there might be changes in the future that break the running of the CI/CD in forks again. An alternative to an automated test would be a list of manual things that need to be checked for each PR, like that the workflows run properly on a fork. That might get out of hand quickly

**Pull request merged**